### ISSUE TITLE
TypeError: scriptElm is undefined

### ISSUE THREAD

**Issue by vandres** [ionic-team/stencil#2500]
**Stencil version:**

```
 @stencil/core@1.14.0
```

**I'm submitting a:**

[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**

I am using Stencil just for components. I include my scripts in a server side technology (Twig) to bind the asset cache to the template cache there:

```
<script type="module" src="/static/js/voan/voan.esm.{{ versionString }}js"></script>
<script nomodule src="/static/js/voan/voan.{{ versionString }}js"></script>
```

Then there is an .htaccess, which rewrites these files to the normal filename.

Having this setup, I always get the error message `TypeError: scriptElm is undefined`, which comes from this code:

```
const scriptElm = BUILD.scriptDataOpts || BUILD.safari10 || BUILD.dynamicImportShim
        ? Array.from(doc.querySelectorAll('script')).find(s => new RegExp(`\/${NAMESPACE}(\\.esm)?\\.js($|\\?|#)`).test(s.src) || s.getAttribute('data-stencil-namespace') === NAMESPACE)
        : null;
    const importMeta = "";
    const opts = BUILD.scriptDataOpts ? scriptElm['data-opts'] || {} : {};
    if (BUILD.safari10 && 'onbeforeload' in scriptElm && !history.scrollRestoration /* IS_ESM_BUILD */) {
```

Is there anyway to include these JS files the way I do? In Version 1.9.2 this technique was working.

**Comment by KevinCarnaille2** [comment]
I don't know if you fixed your problem (hope so) but for other people facing this kind of issue : 
The reason : the RegExp ``\/${NAMESPACE}(\\.esm)?\\.js($|\\?|#)` `doesn't match your script src, as you have added a `versionString` in the middle of the whole src attribute.

If you look closely into the snippet vandres gave us, you have a fallback (not officially documented ?) : 

`... || s.getAttribute('data-stencil-namespace') === NAMESPACE`

For me, adding the `data-stencil-namespace="your-namespace" `attribute to my script tag fixed the problem.

**Comment by rwaskiewicz** [comment]
Hey folks 👋 

I apologize that it took so long for someone on the team to respond to this issue.

There are two ways `scriptElm` could be `undefined`||`null`. I believe we patched the case where `scriptElm` could be `null` in https://github.com/ionic-team/stencil/pull/4101. However, I do believe that @KevinCarnaille2 is correct here & has an open issue to get `'data-stencil-namespace'` documented. 

Given this took so long for the team to respond to, if this is still an issue, and anyone is willing/able to create a minimal reproduction case for the team to take a look at, we'd greatly appreciate it. In the mean time, I've tagged https://github.com/ionic-team/stencil-site/issues/777 to get it ingested into our backlog.

Thanks!
